rules/otbn.bzl

```diff
@@ -260,3 +300,20 @@ otbn_sim_test = rv_rule(
     toolchains = ["@rules_cc//cc:toolchain_type"],
     incompatible_use_toolchain_transition = True,
 )
+
+otbn_consttime_test = rule(
+    implementation = _otbn_consttime_test,
+    test = True,
+    attrs = {
+        "srcs": attr.label_list(allow_files = True),
+        "deps": attr.label_list(providers = [OutputGroupInfo]),
+        "subroutine": attr.string(),
+        "secrets": attr.string_list(),
+        "_checker": attr.label(
+            default = "//hw/ip/otbn/util:check_const_time.py",
```

Comment:
@cfrantz @drewmacrae Is this reasonable? It works but I don't think I'm doing this quite the right way.

I think this should be `//hw/ip/otbn/util:check_const_time` (the `py_binary` target rather than the source file), but I can't get that to work. Here's some things I tried (after copying over `hw/ip/otbn/util/shared/BUILD` from https://github.com/lowRISC/opentitan/pull/13906 because of some missing Python deps):

First, if the `.py` is removed, I get an error saying the `check_const_time` executable can't find `check_const_time.py`, which is fixed by editing the runfiles to include the `.py`:
```diff
 -    runfiles = ctx.runfiles(files = ([ctx.executable._checker, elf]))
+    runfiles = ctx.runfiles(files = (ctx.files._checker + [elf]))
```
However, at that point I get:
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/jadep/.cache/bazel/_bazel_jadep/87e97e71d7aca067127234428526e37b/sandbox/linux-sandbox/1985/execroot/lowrisc_opentitan/bazel-out/k8-fastbuild/bin/sw/otbn/crypto/p256_isoncurve_consttime.sh.runfiles/python3_x86_64-unknown-linux-gnu/bin/python3'
```
Looks like it can't find python3 at all. I couldn't figure out how to indicate that this executable requires Python to be installed.

An alternative I tried was to use a `py_test` rule with the `check_const_time.py` file in `srcs` and `$(location <otbn_file>.elf` as one of the arguments, but this is both very verbose and still ineffective, since Bazel then tells me that the `.elf` file is "not a declared prerequisite for this target" (even if I add the target to `data`). I can't add the target generating the `.elf` file to `deps` because it's not Python.

Maybe it's the case that Bazel just doesn't really allow passing generated-file-dependent arguments to a Python binary inside a test -- if that's the case, I could adjust the constant-time checker to be a library instead of an executable script, and then write a small Python executable inside the test rule that imports this library and runs the test with the provided arguments. This will be my fallback if needed, but it would be nice to run the script directly if possible.

Reply:
In case it helps clarify, here's what a command-line invocation of the test looks like:
```shell
$ hw/ip/otbn/util/check_const_time.py bazel-out/k8-fastbuild-ST-97f470ee3b14/bin/sw/otbn/crypto/p256_ecdsa.elf --verbose --subroutine p256_isoncurve --secrets w2 w3 w4
Analyzing subroutine p256_isoncurve with initial secrets ['w2', 'w3', 'w4']
PASS
```
The exit code is 0 if the test passed, otherwise 1.

Reply:
I think the issue is taht the target of this attribute `//hw/ip/otbn/util:check_const_time` doesn't build.

That target references a handful of targets in `//hw/ip/otbn/util/shared` which do not have build rules (e.g. there is no `py_library` for //hw/ip/otbin/util/shared:{check,control_flow,decode,information_flow_analysis}).  I think if you add those so that `check_const_time` build, then this will be good to go.

Reply:
I added a new commit that pulls in the BUILD file from #13906 so that the target builds, removed `allow_files` and updated the runfiles as I described above. But it doesn't fix the issue; I still get the "can't find python3" error from the earlier comment.